### PR TITLE
Add a manual fatch once in 12 hours

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-29065

## Description
In three API modules, we employ a similar mechanism to prevent fetching feeds with no new data. We utilize the `last_modified` and/or `etag` in the headers to request data only if it has changed since the last retrieval. However, in certain resources, the data may remain unchanged for an extended period, or it may change without updating the `last_modified` field. This situation leads to the expiration of indicators, as their expiration date extends every time the indicator is encountered. In such cases, indicators expire because they haven't been seen for too long due to the absence of new data.

This pull request addresses the issue by implementing a solution: it enforces a fetch with all the data once every 12 hours if a real fetch hasn't occurred within that timeframe. This approach ensures that indicators are updated regularly, preventing unnecessary expiration due to prolonged periods without data retrieval.

We need a force merge because the following tests on the master branch are failing, unrelated to my changes.

This will need a force merge, since the test coverage of CommonServerPython was lower than 80%.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/iDefense/pack_metadata.json
[the remark this answers is not in the document]
Comment (kgal-pan) [reply Packs/iDefense/pack_metadata.json:1]: This Pack is deprecated.

File: Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py [hunk 1]
```diff
@@ -250,14 +260,17 @@ def get_no_update_value(response: requests.models.Response, url: str) -> bool:
 
     etag = response.headers.get('ETag')
     last_modified = response.headers.get('Last-Modified')
+    current_time = datetime.utcnow()
+    # Save the current time as the last updated time. This will be used to check if the indicators have been updated in XSOAR.
+    last_updated = current_time.strftime("%a, %d %b %Y %H:%M:%S GMT")
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py:265]: fixed

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 2]
```diff
@@ -11350,6 +11350,22 @@ def replace_spaces_in_credential(credential):
     return credential
 
 
+def has_passed_time_threshold(timestamp_str, hours_threshold):
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:11353]: Done.

**Review by JasBeilin — COMMENTED** [review]
I am not sure about the new method in CSP. Lets talk about it.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 3]
```diff
@@ -11350,6 +11350,31 @@ def replace_spaces_in_credential(credential):
...
+    Args:
+        timestamp_str (str): The timestamp string in the folloing format: '%Y-%m-%dT%H:%M:%SZ'
+        hours_threshold (int): The threshold in hours.
+        seconds_threshold (int): The threshold in seconds.
+        if both are given, the hours_threshold will be used.
+    Returns:
+        boolean: True if the threshold has passed, False otherwise.
+    """
+    if not hours_threshold and not seconds_threshold:
+        raise ValueError('Either hours_threshold or seconds_threshold must be provided.')
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:11365]: fixed

File: Packs/Base/ReleaseNotes/1_32_53.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+#### Scripts
+
+##### CommonServerPython
+
+- Added a utility methods to determine the time gap between a given datetime and the present moment.
```
Comment (ShirleyDenkberg) [inline Packs/Base/ReleaseNotes/1_32_53.md:6]:
```suggestion
Added a utility method to determine the time gap between a given datetime and the present moment.
```

**Comment by ShirleyDenkberg** [comment]
@kgal-pan @JasBeilin Doc review completed.

**Review by JasBeilin — CHANGES_REQUESTED** [review]
Looks great, please address my comments.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 5]
```diff
@@ -11366,6 +11366,26 @@ def replace_spaces_in_credential(credential):
...
+    :type timestamp_str: ``str``
+    :param timestamp_str: The timestamp to compare the current time to.
+    :type seconds_threshold: ``int``
+    :param seconds_threshold: The threshold in seconds.
+    
+    :return: True if the time difference is greater than the threshold, otherwise False.
+    :rtype: ``bool``
+    """
+
+    timestamp = dateparser.parse(timestamp_str,['%Y-%m-%dT%H:%M:%SZ'])
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:11382]: fixed

**Review by JasBeilin — APPROVED** [review]
Looks good, see my comments.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 6]
```diff
@@ -11366,6 +11366,27 @@ def replace_spaces_in_credential(credential):
...
+    :param timestamp_str: The timestamp to compare the current time to.
+    :type seconds_threshold: ``int``
+    :param seconds_threshold: The threshold in seconds.
+    
+    :return: True if the time difference is greater than the threshold, otherwise False.
+    :rtype: ``bool``
+    """
+    import pytz
+    to_utc_timestamp = dateparser.parse(timestamp_str, settings={'TIMEZONE': 'UTC'})
+    # using astimezone since utcnow() returns a naive datetime object
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:11383]: fixed

**Pull request merged**